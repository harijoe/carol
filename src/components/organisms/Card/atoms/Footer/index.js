import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Wrapper = styled.footer`
  margin-top: auto;
`

const Footer = ({ className, ...props }) =>
  <Wrapper className={`qs-Card-footer ${className}`} {...props} />

Footer.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
}

export default Footer
