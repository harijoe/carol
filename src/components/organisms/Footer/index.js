import React from 'react'
import PropTypes from 'prop-types'

import { FooterAdvices, FooterCorporate, FooterSocialNetworks, MainWrapper } from 'components'

const Footer = ({ locale }) => (
  <MainWrapper>
    <FooterAdvices locale={locale} />
    <FooterCorporate locale={locale} />
    <FooterSocialNetworks locale={locale} />
  </MainWrapper>
)

Footer.propTypes = {
  locale: PropTypes.string,
}

export default Footer
