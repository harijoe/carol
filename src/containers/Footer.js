import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { fromContext, fromRouting } from 'store/selectors'

import { Footer } from 'components'

const FooterContainer = props => <Footer {...props} />

FooterContainer.propTypes = {
  locale: PropTypes.string,
}

const mapStateToProps = state => ({
  locale: fromContext.getLocale(state),
  homepage: fromRouting.isHomepage(state),
})

export default connect(mapStateToProps)(FooterContainer)
