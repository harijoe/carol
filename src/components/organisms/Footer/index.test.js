import React from 'react'
import { shallow } from 'enzyme'

import mockIntl from '../../../../test/intlMock'
import Footer from './'

it('renders the Footer component', () => {
  const wrapper = shallow(mockIntl(<Footer />))

  expect(wrapper).toMatchSnapshot()
})
