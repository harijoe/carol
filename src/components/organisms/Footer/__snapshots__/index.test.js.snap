// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Footer component 1`] = `<Footer />`;
